workspaces/ballerina/ballerina-extension/src/features/ai/service/libs/langlibs.ts

```diff
@@ -14,7 +14,240 @@
+Ballerina provides built-in language libraries (langlibs) for common operations on basic types. These functions are available without explicit imports for default langlibs. Use them whenever you need to perform type conversions, string manipulation, collection operations, or data validation.
 
+## Core Types
+
+**any**: All values except errors. Use for unknown types.
+**anydata**: Serializable data (nil, boolean, int, float, decimal, string, xml, arrays, maps, tables, records). Use for JSON/network/storage. Excludes: error, function, object, typedesc, handle.
+**readonly**: Immutable values (read-only bit on). Use \`readonly & Type\` for immutability.
+- Inherently immutable: nil, boolean, int, float, decimal, string, error, function, typedesc, handle
+- Selectively immutable: xml, arrays, maps, tables, objects
+**Cloneable**: \`readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>\`. Deep-copyable values.
+
+## lang.value
+
+**clone(v)**: Deep copy without copying immutable parts. Use for concurrent modification of structures.
+\`\`\`ballerina
+int[] copy = [1,2,3].clone();
+\`\`\`
+
+**cloneReadOnly(v)**: Immutable deep copy. Use for thread safety.
+\`\`\`ballerina
+int[] & readonly imm = arr.cloneReadOnly();
+\`\`\`
+
+**cloneWithType(v, typedesc)**: Convert and clone to different type. Errors on cycles, converts numerics, applies defaults.
+\`\`\`ballerina
+int[] nums = check anydata[1,2,3].cloneWithType();
+\`\`\`
+
+**ensureType(v, typedesc)**: Safe type cast (returns error vs panic). Use with json/dynamic data.
+\`\`\`ballerina
+string name = check jsonData.name.ensureType();
+\`\`\`
+
+**toString(v)**: Any value to string (direct style). Use for display/logging.
+**toJson(v)**: anydata to json. Use for API compatibility.
+
+**toJsonString(v)**: anydata to JSON string. Use for serialization.
+\`\`\`ballerina
+scores.toJsonString() => '{"Alice":90}'
+\`\`\`
+
+**fromJsonString(str)**: Parse JSON string to json. Numbers → decimal (except ints in range or -0.0).
+\`\`\`ballerina
+json data = check "{\"id\":12}".fromJsonString();
+\`\`\`
+
+**fromJsonWithType(json, typedesc)**: json to specific type. Type-safe deserialization.
+\`\`\`ballerina
+int[] nums = check jsonArr.fromJsonWithType();
+\`\`\`
+
+**fromJsonStringWithType(str, typedesc)**: JSON string → typed value (one-step).
+\`\`\`ballerina
+int[] nums = check "[1,2,3]".fromJsonStringWithType();
+\`\`\`
+
+## lang.array
```

Comment:
Like we discussed, Lets add line based instructions with proper examples for regex and dependent types